File: algorithms/1_easy/C/03042_count_prefix_and_suffix_pairs_i/solution.cpp
class Solution {
public:
    int countPrefixSuffixPairs(vector<string>& words) {
        int count {0};
        const size_t len {words.size()};

        for (size_t i {0}; i < len; ++i) {
            for (size_t j {i + 1}; j < len; ++j) {
                if (words[j].starts_with(words[i]) && words[j].ends_with(words[i])) {
                    count += 1;
                }
            }
        }

        return count;
    }
};

File: algorithms/1_easy/C/03042_count_prefix_and_suffix_pairs_i/solution.rs
impl Solution {
    pub fn count_prefix_suffix_pairs(words: Vec<String>) -> i32 {
        let mut count = 0;
        let len = words.len();

        for i in 0..len {
            for j in (i + 1)..len {
                if words[j].starts_with(&words[i]) && words[j].ends_with(&words[i]) {
                    count += 1;
                }
            }
        }

        count
    }
}
